src/core/dev_api/openvino/core/log_util.hpp

```diff
@@ -10,11 +10,34 @@
 #include "openvino/core/core_visibility.hpp"
 #include "openvino/core/node.hpp"
 
-#ifdef ENABLE_OPENVINO_DEBUG
-
 namespace ov {
 namespace util {
 
+/** @brief Allows toggling log message output between std::cout and user provided callback.
+ */
+class OPENVINO_API LogDispatch {
+public:
+    using Callback = std::function<void(std::string_view)>;
+
+    static Callback* get_callback();
```

Comment:
Don't allow return null pointer callback if we always have callback inside (default or user) there will be less checks

Reply:
Done.